source/tests/system/nirfmxnr_session_tests.cpp

```diff
@@ -131,6 +132,31 @@ TEST_F(NiRFmxNRSessionTest, TwoInitializedSessionsOnSameDevice_CloseSessions_Clo
   EXPECT_EQ(0, close_response_two.status());
 }
 
+TEST_F(NiRFmxNRSessionTest, CallInitializeTwiceWithSameSessionNameOnSameDevice_CloseSessionTwice_SecondCloseReturnsInvalidSessionError)
```

Comment:
This new test is intended to cover Case 1 in the [Session Management part of the HLD](https://ni.visualstudio.com/DevCentral/_git/ASW?path=/Specs/Hardware%20Configuration/E1665864%20gRPC%20API%20for%20RFmx/F1433350%20gRPC%20API%20for%20RFmx%20SpecAn.md&_a=preview&anchor=session-management).

Though, I think it can be debated that the pre-existing unit test covers this: [NamedSessionAdded_AddSessionWithSameName_ReturnsFirstSessionIdAndDoesNotCallInit](https://github.com/ni/grpc-device/blob/a66a53e2cb5bb4faef77bfdc39041ba57b32aefb/source/tests/unit/session_repository_tests.cpp#L91).